### PR TITLE
Introducing runsOn property in task.json, will be used to filter task…

### PR DESCRIPTION
**Pull request by sureshtadisetty**
…s eligible on Agent/MachineGroup/Server.

Valid value of runsOn is list of one or more of these 3 strings [ "Agent", "MachineGroup", "Server" ].
If no value is provided, task is considered eligible to run on Agent (value [ "Agent" ].

### CODE REVIEW

File: Tasks/ANT/task.json [hunk 1]
```diff
@@ -9,10 +9,13 @@
         "Build",
         "Release"
     ],
+    "runsOn": [
+        "Agent"
```
Comment (kmkumaran) [inline Tasks/ANT/task.json]: Is this mandatory in all tasks? If so, how do we handle tasks authored through extensions? They may not have it.

Comment (bryanmacfarlane) [inline Tasks/ANT/task.json]: It shouldn't be.  Default is agent.

File: Tasks/ANT/task.json [hunk 2]
```diff
@@ -9,10 +9,13 @@
...
+    ],
     "author": "Microsoft Corporation",
     "version": {
         "Major": 1,
-        "Minor": 0,
+        "Minor": 1,
```
Comment (bryanmacfarlane) [inline Tasks/ANT/task.json]: I would avoid rev'ing all tasks just to add runsOn Agent.  It should default to that.  

Don't require the server to store and agents to redownload for no reason.

Comment (sureshtadisetty) [inline Tasks/ANT/task.json]: Sure, I'll undo changes for all Agent only tasks.

File: Tasks/IISWebAppDeployment/task.json [hunk 3]
```diff
@@ -10,10 +10,14 @@
     "Build",
     "Release"
   ],
+  "runsOn": [
+    "Agent",
```
Comment (kmkumaran) [inline Tasks/IISWebAppDeployment/task.json]: This task is deprecated. You can ignore this file.

File: Tasks/SqlServerDacpacDeployment/task.json [hunk 4]
```diff
@@ -10,10 +10,14 @@
     "Build",
     "Release"
   ],
+  "runsOn": [
```
Comment (kmkumaran) [inline Tasks/SqlServerDacpacDeployment/task.json]: This file is also deprecated.

**Comment by kmkumaran** [comment]
Looks fine.

**Comment by bryanmacfarlane** [comment]
close or take this ASAP

**Comment by kmkumaran** [comment]
@sureshtadisetty - I am closing this stale PR.

**Pull request closed without merging**